cylc/uiserver/tests/test_gui.py

```diff
@@ -13,59 +13,153 @@
             id='existing_workflow_new_workflow'
         ),
         pytest.param(
-            'content="1;url=http://localhost:8892/cylc/?token=1234567890some_big_long_token1234567890#/workspace/some/workflow" /> ',
+            'http://localhost:8892/cylc/?token=1234567890some_big_long_token1234567890#/workspace/some/workflow',
             None,
-            'content="1;url=http://localhost:8892/cylc/?token=1234567890some_big_long_token1234567890#" /> ',
+            'http://localhost:8892/cylc/?token=1234567890some_big_long_token1234567890#',
             id='existing_workflow_no_new_workflow'
         ),
         pytest.param(
-            'content="1;no url in this file "',
+            '',
             'another/flow',
-            'content="1;no url in this file "',
+            None,
             id='no_url_no_change'
         ),
     ]
 )
 def test_update_html_file_updates_gui_file(
     existing_content,
     workflow_id,
-    expected_updated_content,
-    tmp_path):
-    """Tests html file is updated correctly"""
+        expected_updated_content):
+    """Tests url is updated correctly"""
+
+    updated_file_content = update_url(existing_content, workflow_id)
+    assert updated_file_content == expected_updated_content
+
+
+@pytest.mark.parametrize(
+    'file_content,expected_url',
+    [
+        pytest.param(
+            'content="1;url=http://localhost:8892/cylc/?token=1234567890some_big_long_token1234567890#/workspace/some/workflow" /> ',
+            "http://localhost:8892/cylc/?token=1234567890some_big_long_token1234567890#/workspace/some/workflow",
+            id='url_in_file'
+        ),
+        pytest.param(
+            'There is no url in here',
+            None,
+            id='no_url_in_file'
+        ),
+    ]
+)
+def test_get_url_from_file(file_content, expected_url, tmp_path):
     Path(tmp_path).mkdir(exist_ok=True)
     tmp_gui_file = Path(tmp_path / "gui")
     tmp_gui_file.touch()
-    tmp_gui_file.write_text(existing_content)
-    update_html_file(tmp_gui_file, workflow_id)
-    updated_file_content = tmp_gui_file.read_text()
+    tmp_gui_file.write_text(file_content)
+    actual_url = get_url_from_file(tmp_gui_file)
+    assert actual_url == expected_url
+    rmtree(tmp_path, ignore_errors=True)
 
-    assert updated_file_content == expected_updated_content
+
+def test_gui_selection_and_clean_process(tmp_path, monkeypatch):
+    """Testing functionally the gui selection and cleaning process"""
+    # set up file structure
+    info_files_dir = Path(tmp_path/'.cylc'/'uiserver'/'info_files')
+    info_files_dir.mkdir(parents=True, exist_ok=True)
+    for i in range(1, 5):
+        pid = randint(1000, 100000)
+        html_file = (info_files_dir / f"jpserver-{pid}-open.html")
+        # the json file is unused but created empty to ensure the html is the
+        # file used for selection
+        _json_file = (info_files_dir / f"jpserver-{pid}.json")
```

Comment:
This is not actually being created in the test
```suggestion
        json_file = (info_files_dir / f"jpserver-{pid}.json")
        json_file.touch()
```